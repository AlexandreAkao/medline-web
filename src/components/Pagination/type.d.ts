type ReactPaginateProps = import('react-paginate').ReactPaginateProps;

interface IPaginationProps extends ReactPaginateProps {
  pageCount: number;
  onPageChange?: (selectedItem: { selected: number }) => void;
  previousLabel?: string;
  nextLabel?: string;
  className?: string;
}
