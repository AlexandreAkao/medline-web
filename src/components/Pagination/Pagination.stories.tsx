import { ComponentStory, ComponentMeta } from '@storybook/react';

import Pagination from 'components/Pagination';
import colors from 'styles/colors';

export default {
  title: 'Pagination',
  args: {},
  component: Pagination,
  decorators: [
    Story => (
      <div
        style={{
          background: colors.primary.normal,
          width: '100%',
          height: '100%',
        }}
      >
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Pagination>;

const Template: ComponentStory<typeof Pagination> = args => <Pagination {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  pageCount: 10,
};
