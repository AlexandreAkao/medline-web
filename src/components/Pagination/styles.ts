import styled from 'styled-components';

import colors from 'styles/colors';

export const PaginationContainer = styled.div`
  box-shadow: 0px 0px 20px 0px rgba(0, 0, 0, 0.2);
  border-radius: 1rem;

  .pagination-container {
    display: flex;

    li {
      font-size: 1.125rem;
      cursor: pointer;
      background-color: ${colors.white.lightest};
      color: ${colors.black.soft};

      a {
        display: inline-flex;
        padding: 1rem;
      }
    }

    .selected {
      background-color: ${colors.primary.lighter};
      color: ${colors.primary.normal};
      font-weight: 600;
      border: 1px;
    }

    .previous {
      border-radius: 1rem 0 0 1rem;
    }

    .next {
      border-radius: 0 1rem 1rem 0;
    }

    .disabled {
      cursor: auto;
    }
  }
`;
