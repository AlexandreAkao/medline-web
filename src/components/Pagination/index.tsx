import ReactPaginate from 'react-paginate';

import { PaginationContainer } from 'components/Pagination/styles';

function Pagination({
  pageCount,
  previousLabel = '<',
  nextLabel = '>',
  className = 'pagination-container',
  onPageChange,
  ...rest
}: IPaginationProps) {
  return (
    <PaginationContainer>
      <ReactPaginate
        {...rest}
        previousLabel={previousLabel}
        nextLabel={nextLabel}
        className={className}
        pageCount={pageCount}
        onPageChange={onPageChange}
      />
    </PaginationContainer>
  );
}

export default Pagination;
