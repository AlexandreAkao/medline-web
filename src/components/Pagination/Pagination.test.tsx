import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';
import { vi } from 'vitest';

import * as stories from 'components/Pagination/Pagination.stories';
import Pagination from 'components/Pagination';

const { Primary } = composeStories(stories);

vi.mock('components/Pagination', async () => {
  return {
    default: () => <div>Pagination</div>,
  };
});

describe('Pagination', () => {
  const renderComponent = (customProps: Partial<IPaginationProps> = {}) => {
    const props: IPaginationProps = {
      pageCount: 10,
      ...customProps,
    };

    return render(<Pagination {...props} />);
  };

  it('should match with snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render a primary Pagination', () => {
    render(<Primary />);
    const labelElement = screen.getByText(/Pagination/i);
    expect(labelElement).toBeInTheDocument();
  });
});
