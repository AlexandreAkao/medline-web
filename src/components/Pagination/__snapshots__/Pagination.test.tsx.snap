// Vitest Snapshot v1

exports[`Pagination > should match with snapshot 1`] = `
<div>
  <div>
    Pagination
  </div>
</div>
`;
